File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b);
    let answer = 0;
    
    let [i, j] = [0, people.length - 1];
    while(i <= j) {
        if(people[i] + people[j] <= limit) i++;
        j--;
        answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    n = len(people)
    people.sort()
    left, right = 0, n - 1
    twin = 0
    while left < right:
        if people[left] + people[right] <= limit:
            twin += 1
            left += 1
            right -= 1
        else:
            right -= 1
    return n - twin
            